herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -1667,10 +1691,10 @@ private void applyInsert(Bytes key, Bytes value, boolean onTransaction) throws D
 
         if (table.auto_increment) {
             // the next auto_increment value MUST be greater than every other explict value
-            long pk_logical_value;
-            if (table.getColumn(table.primaryKey[0]).type == ColumnTypes.INTEGER) {
+            long pk_logical_value = key.to_long();
```

Comment:
in case of an int you have to use key.to_int(), this will result in some bad runtime error (maybe the error in your test)
because you can convert only a 8-bytes Bytes into a long, not a 4-bytes one

Reply:
Yes. Good catch. I will fix this.